drivers/vulkan/rendering_device_driver_vulkan.cpp

```diff
@@ -5178,6 +5520,236 @@ RDD::PipelineID RenderingDeviceDriverVulkan::render_pipeline_create(
+#endif
+}
+
+// ----- COMMANDS -----
+
+void RenderingDeviceDriverVulkan::command_build_acceleration_structure(CommandBufferID p_cmd_buffer, AccelerationStructureID p_acceleration_structure) {
+#if !(defined(MACOS_ENABLED) || defined(IOS_ENABLED))
+	const AccelerationStructureInfo *accel_info = (const AccelerationStructureInfo *)p_acceleration_structure.id;
+	const VkAccelerationStructureBuildRangeInfoKHR *range_info_ptr = &accel_info->range_info;
+	vkCmdBuildAccelerationStructuresKHR((VkCommandBuffer)p_cmd_buffer.id, 1, &accel_info->build_info, &range_info_ptr);
+#endif
+}
+
+void RenderingDeviceDriverVulkan::command_bind_raytracing_pipeline(CommandBufferID p_cmd_buffer, RaytracingPipelineID p_pipeline) {
+	vkCmdBindPipeline((VkCommandBuffer)p_cmd_buffer.id, VK_PIPELINE_BIND_POINT_RAY_TRACING_KHR, (VkPipeline)p_pipeline.id);
+}
+
+void RenderingDeviceDriverVulkan::command_bind_raytracing_uniform_set(CommandBufferID p_cmd_buffer, UniformSetID p_uniform_set, ShaderID p_shader, uint32_t p_set_index) {
+	const ShaderInfo *shader_info = (const ShaderInfo *)p_shader.id;
+	const UniformSetInfo *usi = (const UniformSetInfo *)p_uniform_set.id;
+	vkCmdBindDescriptorSets((VkCommandBuffer)p_cmd_buffer.id, VK_PIPELINE_BIND_POINT_RAY_TRACING_KHR, shader_info->vk_pipeline_layout, p_set_index, 1, &usi->vk_descriptor_set, 0, nullptr);
+}
+
+void RenderingDeviceDriverVulkan::command_raytracing_trace_rays(CommandBufferID p_cmd_buffer, RaytracingPipelineID p_pipeline, ShaderID p_shader, uint32_t p_width, uint32_t p_height) {
+#if !(defined(MACOS_ENABLED) || defined(IOS_ENABLED))
+	ShaderInfo *shader_info = (ShaderInfo *)p_shader.id;
+
+	uint32_t handle_size = raytracing_capabilities.shader_group_handle_size;
+	uint32_t handles_size = shader_info->regions.handles_data.size();
+	uint8_t *handles_ptr = shader_info->regions.handles_data.ptr();
+
+	VkResult err = vkGetRayTracingShaderGroupHandlesKHR(vk_device, (VkPipeline)p_pipeline.id, 0, shader_info->regions.group_count, handles_size, handles_ptr);
+	ERR_FAIL_COND_MSG(err, "vkGetRayTracingShaderGroupHandlesKHR failed with error " + itos(err) + ".");
+
+	uint8_t *sbt_ptr = buffer_map(shader_info->sbt_buffer);
+	uint8_t *sbt_data = sbt_ptr;
+	uint32_t handle_index = 0;
+
+	memcpy(sbt_data, handles_ptr + handle_index * handle_size, handle_size);
+	++handle_index;
+
+	sbt_data = sbt_ptr + shader_info->regions.raygen.size;
+	for (uint32_t i = 0; i < shader_info->regions.miss_count; ++i) {
+		memcpy(sbt_data, handles_ptr + handle_index * handle_size, handle_size);
+		sbt_data += shader_info->regions.miss.stride;
+		++handle_index;
+	}
+
+	sbt_data = sbt_ptr + shader_info->regions.raygen.size + shader_info->regions.miss.size;
+	for (uint32_t i = 0; i < shader_info->regions.closest_hit_count; ++i) {
+		memcpy(sbt_data, handles_ptr + handle_index * handle_size, handle_size);
+		sbt_data += shader_info->regions.closest_hit.stride;
+		++handle_index;
+	}
+
+	buffer_unmap(shader_info->sbt_buffer);
```

Comment:
If this data was filled out and remains static during the lifetime of the shader, then it must be mapped and fill during creation and only then. If this varies per dispatch call, then it'll completely overwrite the data used by the previous call. This seems to indicate to me this is in the wrong spot.

As far as I can see, it currently seems to indicate that this depends on the pipeline itself, which means the buffer might belong there and not in the ShaderInfo and can probably be created and filled right after the raytracing pipeline is created.